internal/services/datafactory/data_factory_dataset_binary_resource.go

```diff
@@ -229,16 +230,21 @@ func resourceDataFactoryDatasetBinary() *pluginsdk.Resource {
 						},
 						// SFTP Specific field
 						"type": {
-							Type:     pluginsdk.TypeString,
-							Required: true,
+							Type:             pluginsdk.TypeString,
+							Required:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
this indicates a bug in the API that needs to be resolved/worked around above - so we need to do that and remove this here:

```suggestion
```

Reply:
Both "Gzip" and "gzip" is working, I think it's safe to remove this func